app/js/handlebarsScoreTable.js

```diff
@@ -20,12 +20,28 @@ async function sortUsersObjectByDate() {
  * with user objects and send this to searching and filtering.
  */
 function updateScoreTable() {
-    let users = sortUsersObjectByDate();
+    let users = sortUsersObjectByDate()
     users.then(function (userInfo) {
         getTemplateAjax('js/templates/adminTable.hbs').then(function (HBTemplate) {
-            let filteredUserArray = searchAndFilter(userInfo.data);
-            updateChart(filteredUserArray);
-            printFilteredResultsToScreen(HBTemplate, filteredUserArray);
+            let filteredUserArray = searchAndFilter(userInfo.data)
+            updateChart(filteredUserArray)
+            printFilteredResultsToScreen(HBTemplate, filteredUserArray)
+            document.querySelectorAll('.download-user-results-button').forEach((button) => {
+                button.addEventListener('click', (e) => {
+                    e.preventDefault()
+
+                    getData("result?id=" + e.target.parentElement.getAttribute("dataId"))
+
+                    .then((data) => {
+                        let parentElement = e.target.parentElement
+                        let userId = parentElement.getAttribute("dataId")
+                        let userName = parentElement.getAttribute("dataname")
+                        let userPercentage = parentElement.getAttribute("datapercentage")
+                        let result = data
+                        downloadFile(`${userName}_aptitude_test_results`, createCSV(result, userId, userName, userPercentage))
+                    })
+                })
+            })
```

Comment:
Why is this in here? Update score table now also registers download button event listeners?

Reply:
This should be in it's own function that gets called here instead